compiler/next/include/chpl/resolution/scope-queries.h

```diff
@@ -46,6 +51,90 @@ namespace resolution {
   const ResolvedVisibilityScope* resolveVisibilityStmts(Context* context,
                                                         const Scope* scope);
 
+  /**
+    Given an Expression and a Scope, return the things
+    that Expression might refer to.
+
+    * If checkDecls is true, looks for symbols declared in this Scope.
+    * If checkUseImport is true, looks for symbols from use/import statements in
+      this Scope.
+    * If checkParents is true, looks for symbols from parent scopes
+      (but not parent modules of a module) including looking for declarations
+      and handling imports, and including finding declarations in the root
+      module.
+    * If checkToplevel is true, checks for a toplevel module with this name.
+    * If findOne is true, limits search to the innermost scope with a match.
+   */
+  std::vector<BorrowedIdsWithName> lookupInScope(Context* context,
```

Comment:
Yeah. I have been using the strategy of having the call sites use a comment that identifies the individual bools (which is cheap and a practice I apply generally). E.g. `/* checkDecls */ true`.

I'm not really thrilled by just replacing true/false with `CheckDecls::True` etc. I'm not really sure why that doesn't appeal to me but I do see that it would reduce the need to write things like `/* checkDecls */ true` at the call site. I think my objection to it is that it seems to make a simple thing (passing true/false) more complex than it needs to be.

At the same time, I had been wondering if I should bundle up all of these bool arguments into a struct. I find the large number of arguments to these functions annoying. I like the idea of adding a bitfield since it both makes the call sites clearer & bundles them. So I'll look in to that.